src/poetry/console/commands/init.py

```diff
@@ -237,6 +240,30 @@ def handle(self) -> int:
         with (Path.cwd() / "pyproject.toml").open("w", encoding="utf-8") as f:
             f.write(content)
 
+    def _generate_choice_list(
+        self, matches: List["Package"], canonicalized_name: str, raw_name: str
+    ) -> Tuple[List[str], str]:
+        choices = []
+        matches_names = [p.name for p in matches]
+        exact_match = canonicalized_name in matches_names
+        info_string: str = (
+            f"Found <info>{len(matches)}</info> packages matching <c1>{raw_name}</c1>"
+        )
+        if exact_match:
+            choices.append(matches[matches_names.index(canonicalized_name)].pretty_name)
+
+        for found_package in matches:
+            if len(choices) >= 10:
+                info_string += "\nNOTE: Output has been truncated for readability"
+                break
+
+            if found_package.name == canonicalized_name:
+                continue
+
+            choices.append(found_package.pretty_name)
+
+        return choices, info_string
```

Comment:
Hm, we shouldn't create and return this info_string within the method. It would mean that this method is doing more than "one thing". Instead we should keep it there where it was before and just extend the message by a "Showing the first 10 results" or similar in the case of `len(matches) >= 10`.